packages/inertia/src/inertia.js

```diff
@@ -53,6 +67,7 @@ export default {
   visit(url, { method = 'get', data = {}, replace = false, preserveScroll = false, preserveState = false, only = [] } = {}) {
     Progress.start()
     this.cancelActiveVisits()
+    this.saveScrollPositions()
```

Comment:
Is this still necessary since they're being saved on scroll?

Reply:
I think so, lol. I wrote this months ago, and I think there was an edge case reason for this.

Reply:
Maybe I'll remove it, and see if I get any issues reported.

Reply:
Did some digging into this. 

Here's an initial page load, followed by scrolling, clicking a link, and clicking the back button:

![Screenshot 2020-08-20 at 13 40 59](https://user-images.githubusercontent.com/1752195/90765765-c8f5d480-e2ea-11ea-846d-922f799e2053.png)

I also tried causing a race condition by clicking a visit link (emulated 3g), scrolling, and only after hitting the next page. This also worked fine:

![Screenshot 2020-08-20 at 13 43 27](https://user-images.githubusercontent.com/1752195/90766219-7ec12300-e2eb-11ea-9c6e-ee6ce576ac8f.png)

My assumption here (I'll check the changelog after placing this comment) is that the this was done to ensure a state existed, so that if you go back to a previous page, the `restoreState` method doesn't accidentally throw an error.

However, because `setPage`'s nested `updatePage` also has a `saveScrollPositions` call, this doesn't happen any longer, which appears to indeed make this specific call obsolete. 

That said, I wouldn't only rely on storing on-scroll, because it's debounced and therefor doesn't always save (e.g. automated test environment, or if you manage to navigate within those 100ms)

Reply:
Hey thanks @claudiodekker, I appreciate you digging into this.

> That said, I wouldn't only rely on storing on-scroll, because it's debounced and therefor doesn't always save (e.g. automated test environment, or if you manage to navigate within those 100ms)

I'm pretty sure this was my thinking. Since it really hurt to save the scroll position again on visit, it felt like a good idea. And, I feel like I did run into situations where this was actually needed.

Maybe I'll just leave it as it. 😝

Reply:
> Maybe I'll remove it, and see if I get any issues reported.

You madman! 😂

Reply:
> You madman! 😂

I've already changed my mind lol 😅

Reply:
> Hey thanks @claudiodekker, I appreciate you digging into this.

Sure 👍  I'm on my lunch break anyway 😄
Re: my earlier assumption:
> .. so that if you go back to a previous page, the restoreState method doesn't accidentally throw an error.

I just noticed that there's actually already a guard-statement in place, so I guess that this shouldn't happen anyway?: https://github.com/inertiajs/inertia/blob/54d1a413e4a883afa49609b26a8dd3749e5ebb1e/packages/inertia/src/inertia.js#L188

If that works correctly, then there isn't even a need to manually call `this.saveScrollPositions()` other than in the scroll handler :) I'll make some changes on my end, and will see if that works.

Reply:
Pushed my changes to this in a separate PR, as this one was already merged 😅  https://github.com/inertiajs/inertia/pull/192